crates/bevy_ecs/macros/src/lib.rs

```diff
@@ -363,6 +363,41 @@ pub fn derive_system_param(input: TokenStream) -> TokenStream {
     let generics = ast.generics;
     let (impl_generics, ty_generics, where_clause) = generics.split_for_impl();
 
+    let lifetime_generics: Vec<_> = generics
+        .params
+        .iter()
+        .map(|g| match g {
+            GenericParam::Lifetime(l) => Some(l),
+            _ => None,
+        })
+        .flatten()
+        .map(|l| &l.lifetime)
+        .collect();
```

Comment:
Just curious if there is a reason that you opted against using the built-in `.lifetimes()` function like this? I did a quick test and it seems functionally identical.
```suggestion
    let lifetime_generics: Vec<_> = generics.lifetimes().map(|l| &l.lifetime).collect();
```

Reply:
Guess I should have checked the docs, didn't know that one existed. Suggestion applied!